test/TPP/linalg-to-tpp.mlir

```diff
@@ -99,3 +99,29 @@ func.func @main() -> memref<8x32x32x32xf32> {
   }
   return %alloc : memref<8x32x32x32xf32>
 }
+
+// -----
+
+#map = affine_map<(d0) -> (d0)>
+func.func @add_mapping(%arg0: memref<1xf32>, %arg1: memref<1xf32>) {
+  // CHECK: tpp.add
+  linalg.generic {indexing_maps = [#map, #map], iterator_types = ["parallel"]} ins(%arg0: memref<1xf32>) outs(%arg1: memref<1xf32>) {
+    ^bb0(%in: f32, %out: f32):
+      %0 = arith.addf %in, %out : f32
+      linalg.yield %0 : f32
+  }
+  return 
+}
+
+// -----
+
+#map = affine_map<() -> ()>
+func.func @add_mapping(%arg0: memref<f32>, %arg1: memref<f32>) {
+  // CHECK-NOT: tpp.add
```

Comment:
Make sure to add a comment as to why this should fail. someone looking at the test won't necessarily see the commit. Also, add tests for > 2 and all corner cases for the dim=1 escape hatch.